File: minimum_working_example.R
library(reticulate)
library(tidyverse)

setwd("~/repo/ui_calculator/")

#point this to a conda environment that includes numpy and pandas. The YAML export of the environment
#we used is in the source folder.
use_condaenv() #You could also replace this with use_python()


source_python("source/ui_calculator.py")


#Example of weekly benefits calculated
income_data <- read.csv("example_annual.csv")

income_data <- income_data %>% mutate(weekly_earnings = wage/weeks_worked,
                   q1_earnings = weeks_worked - 39,
                   q2_earnings = weeks_worked - 26,
                   q3_earnings = weeks_worked - 13,
                   q4_earnings = weeks_worked) %>%
  mutate_at(vars(matches("q[1-4]_earnings" )), ~ case_when(.x > 13 ~ 13*weekly_earnings,
                                                           .x < 0 ~ 0,
                                                           TRUE ~ .x*weekly_earnings)) %>%
  mutate(benefits_amount = calc_weekly_state_quarterly(q1_earnings,
                                                       q2_earnings,
                                                       q3_earnings,
                                                       q4_earnings,
                                                       state) %>% map_dbl(1))
print(income_data)

File: minimum_working_example.py
# -*- coding: utf-8 -*-
"""
Created on Tue May 12 09:22:20 2020

@author: probert2
"""
import pandas as pd
import numpy as np
import sys
sys.path.insert(0, 'source')

from ui_calculator import calc_weekly_state_quarterly

income_data = pd.read_csv("example_annual.csv")

income_data['weekly_earnings'] = income_data['wage']/income_data['weeks_worked']

def quarterly_earnings(annual_weeks, weekly_earnings, quarter_number):
    quarter_weeks = annual_weeks - 52 + 13 * quarter_number
    
    quarter_weeks = min(max(0, quarter_weeks), 13)
    
    return quarter_weeks*weekly_earnings

v_quarterly_earnings = np.vectorize(quarterly_earnings)

for i in range(4):
    income_data['q' + str(i + 1)] = v_quarterly_earnings(income_data['weeks_worked'], income_data['weekly_earnings'], i + 1)
    

income_data['benefits'] = calc_weekly_state_quarterly(
        income_data['q1'],                      
        income_data['q2'],
        income_data['q3'],
        income_data['q4'],
        income_data['state'],
        income_data['weeks_worked'])
